pages/claim/claim.js

```diff
@@ -0,0 +1,279 @@
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import React, {useState, useEffect} from 'react';
+import Container from "@mui/material/Container";
+import CircularProgress from '@mui/material/CircularProgress';
+import CheckMarkDone from '../../lib/checkMarkDone.js';
+import {setNewRecipient, claimReputation, getRecipient, getPendingTXStatus} from '../../lib/connect.serv.js';
+
+
+const isEth = /^0x[a-fA-F0-9]{40}$/;
+
+/**
+ * Claim component. For setting a new recipient (optional), 
+ * and claim the GOOD tokens for connected network
+ * @param props contains: proofData array, currentConnection Object, callback backToSwitch, 
+ * isMobile boolean
+ * 
+ */
+export default function Claim(props){
+  const [proof, setProof] = useState(null);
+  const [connectionDetails, setConnectionDetails] = 
+        useState({connectedChain: null, connectedAddress: null});
+  const [query, setQuery] = useState({status: 'init'});
+  const [repRecipient, setRepRecipient] = useState(null);
+  const [contractInstance, setContractInstance] = useState(null);
+  const [isMob, setIsMobile] = useState(null);
+  const [newRecValue, setNewRecValue] = useState(null);
+
+  useEffect(() => {
+    setIsMobile(props.isMobile);
+    setProof(props.proofData);
+    getRec(props.currentConnection);
+    setConnectionDetails(props.currentConnection);
+  }, [props]);
+
+  const changeRecipient = async(e) => {
```

Comment:
it is generally a good idea to wrap all functions you define in a "useCallback"
specifically functions that are passed to child components
i've wrote this comment on this function but it is relevant for all